src/model/parseModelDefinition.ts

```diff
@@ -21,8 +22,15 @@ export function parseModelDefinition<Dictionary extends ModelDictionary>(
   dictionary: Dictionary,
   modelName: string,
   definition: ModelDefinition,
+  propertyPath = '',
```

Comment:
I don't think that exposing the internal details like `propertyPath` is a good idea in this function.

The intention behind `parseModelDefinition` is to produce a representation of the model given a dictionary. I understand the technical reasons you're adding this argument (to support recursion) but I think this is done at the wrong level. 

I'd much prefer for the `parseModelDefinition` function to remain as-is and solve the recursive parsing otherwise.

Reply:
done, hope it's better now :)